bigtable-dataflow-parent/bigtable-beam-import/README.md

```diff
@@ -1,44 +1,215 @@
 # HBase Sequence Files to Cloud Bigtable using Beam
 
-This project supports importing and exporting HBase Sequence Files to Google Cloud Bigtable using
-Cloud Dataflow.
+This folder contains tools to support importing and exporting HBase data to
+Google Cloud Bigtable using Cloud Dataflow.
 
-## Instructions
-[//]: # ({x-version-update-start:bigtable-dataflow-parent:released})
-Download [the import/export jar](http://search.maven.org/remotecontent?filepath=com/google/cloud/bigtable/bigtable-beam-import/1.14.1/bigtable-beam-import-1.14.1-shaded.jar), which is an aggregation of all required jars.
+## Setup 
 
-Please pay attention to the Cluster CPU usage and adjust the number of Dataflow workers accordingly.
+To use the tools in this folder, you can download them from the maven repository, or
+you can build them using Maven. 
+
+
+[//]: # ({x-version-update-start:bigtable-dataflow-parent:released})
+### Download the jars
+Download [the import/export jars](http://search.maven.org/remotecontent?filepath=com/google/cloud/bigtable/bigtable-beam-import/1.14.1/bigtable-beam-import-1.14.1-shaded.jar), which is an aggregation of all required jars.
 
-## Export
+### Build the jars yourself
 
-On the command line:
+Go to the top level directory and build the repo
+then return to this sub directory.
 
 ```
-java -jar bigtable-beam-import-1.14.1-shaded.jar export \
-    --runner=dataflow \
-    --project=[your_project_id] \
-    --bigtableInstanceId=[your_instance_id] \
-    --bigtableTableId=[your_table_id] \
-    --destinationPath=gs://[bucket_name]/[export_directory]/ \
-    --tempLocation=gs://[bucket_name]/[temp_work_directory]/ \
-    --maxNumWorkers=[10x number of nodes] \
+cd ../../
+mvn clean install -DskipTests=true
+cd bigtable-dataflow-parent/bigtable-beam-import
 ```
 
-## Import
+***
+#Tools
 
-Create the table in your cluster.
+## Data export pipeline
+
+You can export data into a snapshot or into sequence files. If you're migrating
+your data from HBase to Bigtable using snapshots is the preferred method. 
+
+### Exporting snapshots from HBase
+
+Perform these steps from Unix shell on an HBase edge node.
+
+1. Set the environment variables
+    ```
+    TABLE_NAME=your-table-name
+    SNAPSHOT_NAME=your-snapshot-name 
+    SNAPSHOT_EXPORT_PATH=/hbase-migration-snap
+    BUCKET_NAME="gs://bucket-name"
+    ```
+1. Take the snapshot
+    ```
+    echo "snapshot '$TABLE_NAME', '$SNAPSHOT_NAME'" | hbase shell -n
+    ```
+
+1. Export the snapshot   
+    ```
+    hbase org.apache.hadoop.hbase.snapshot.ExportSnapshot -snapshot $SNAPSHOT_NAME \
+        -copy-to $BUCKET_NAME$SNAPSHOT_EXPORT_PATH/data -mappers 16
+    ```
+1. Create hashes for the table to be used during the data validation step.
+   ```
+   hbase org.apache.hadoop.hbase.mapreduce.HashTable --batchsize=10 --numhashfiles=10 \
```

Comment:
batchsize is number of bytes we scan in computing a single hash. We'd likely set this closer to a kb.  With TB 
 sized tables we should expect the results in the GBs. Instead of providing recommendations here, we should like to hbase doc: http://hbase.apache.org/book.html#_step_1_hashtable

Reply:
I think we should provide this information. I will add the link for now, but let's discuss adding a more thorough guide to the params